packages/editor-ui/src/stores/workflows.store.ts

```diff
@@ -108,1355 +100,1474 @@ const createEmptyWorkflow = (): IWorkflowDb => ({
+			return execution.id === finishedActiveExecution.executionId;
+		});
+
+		if (activeExecutionIndex === -1) {
+			// The execution could not be found
+			return;
+		}
+
+		const activeExecution = activeExecutions.value[activeExecutionIndex];
+
+		activeExecutions.value = [
+			...activeExecutions.value.slice(0, activeExecutionIndex),
+			{
+				...activeExecution,
+				...(finishedActiveExecution.executionId !== undefined
+					? { id: finishedActiveExecution.executionId }
+					: {}),
+				finished: finishedActiveExecution.data.finished,
+				stoppedAt: finishedActiveExecution.data.stoppedAt,
+			},
+			...activeExecutions.value.slice(activeExecutionIndex + 1),
+		];
+
+		if (finishedActiveExecution.data && (finishedActiveExecution.data as IRun).data) {
+			setWorkflowExecutionRunData((finishedActiveExecution.data as IRun).data);
+		}
+	}
+
+	function setActiveExecutions(newActiveExecutions: IExecutionsCurrentSummaryExtended[]): void {
+		activeExecutions.value = newActiveExecutions;
+	}
+
+	// TODO: For sure needs some kind of default filter like last day, with max 10 results, ...
+	async function getPastExecutions(
+		filter: IDataObject,
+		limit: number,
+		lastId?: string,
+		firstId?: string,
+	): Promise<IExecutionsListResponse> {
+		let sendData = {};
+		if (filter) {
+			sendData = {
+				filter,
+				firstId,
+				lastId,
+				limit,
+			};
+		}
+		const rootStore = useRootStore();
+		return await makeRestApiRequest(rootStore.getRestApiContext, 'GET', '/executions', sendData);
+	}
+
+	async function getActiveExecutions(
+		filter: IDataObject,
+	): Promise<IExecutionsCurrentSummaryExtended[]> {
+		let sendData = {};
+		if (filter) {
+			sendData = {
+				filter,
+			};
+		}
+		const rootStore = useRootStore();
+		const output = await makeRestApiRequest<{ results: IExecutionsCurrentSummaryExtended[] }>(
+			rootStore.getRestApiContext,
+			'GET',
+			'/executions',
+			sendData,
+		);
+
+		return output.results;
+	}
+
+	async function getExecution(id: string): Promise<IExecutionResponse | undefined> {
+		const rootStore = useRootStore();
+		const response = await makeRestApiRequest<IExecutionFlattedResponse | undefined>(
+			rootStore.getRestApiContext,
+			'GET',
+			`/executions/${id}`,
+		);
+
+		return response && unflattenExecutionData(response);
+	}
+
+	async function createNewWorkflow(sendData: IWorkflowDataUpdate): Promise<IWorkflowDb> {
+		// make sure that the new ones are not active
+		sendData.active = false;
+
+		const rootStore = useRootStore();
+		return await makeRestApiRequest(
+			rootStore.getRestApiContext,
+			'POST',
+			'/workflows',
+			sendData as unknown as IDataObject,
+		);
+	}
+
+	async function updateWorkflow(
+		id: string,
+		data: IWorkflowDataUpdate,
+		forceSave = false,
+	): Promise<IWorkflowDb> {
+		const rootStore = useRootStore();
+
+		if (data.settings === null) {
+			EventReporter.info('Detected workflow payload with settings as null');
```

Comment:
this was added in https://github.com/n8n-io/n8n/pull/9227 to report this issue to Sentry. but we do not have Sentry setup for the frontend since months. This is dead code, and should be deleted.

Reply:
Good point. Deleted!